cloud/askv.go

```diff
@@ -2,31 +2,53 @@ package cloud
 
 import (
 	"context"
+	"strings"
 
 	"github.com/earthly/cloud-api/askv"
 	"github.com/pkg/errors"
 )
 
-func (c *Client) AutoSkipExists(ctx context.Context, org, project, pipeline string, hash []byte) (bool, error) {
+func (c *Client) AutoSkipExists(ctx context.Context, org, project string, hash []byte) (bool, error) {
 	response, err := c.askv.Exists(c.withAuth(ctx), &askv.ExistsRequest{
 		OrgName:     org,
 		ProjectName: project,
 		Hash:        hash,
 	})
 	if err != nil {
-		return false, errors.Wrap(err, "failed querying auto-skip service")
+		return false, errors.Wrap(err, "auto-skip existence check failed")
 	}
 	return response.Exists, nil
 }
 
-func (c *Client) AutoSkipAdd(ctx context.Context, org, project, pipeline string, hash []byte) error {
+func (c *Client) AutoSkipAdd(ctx context.Context, org, project, path, target string, hash []byte) error {
 	_, err := c.askv.Add(c.withAuth(ctx), &askv.AddRequest{
 		OrgName:     org,
 		ProjectName: project,
+		TargetName:  target,
+		TargetPath:  path,
 		Hash:        hash,
 	})
 	if err != nil {
-		return errors.Wrap(err, "failed adding to auto-skip service")
+		return errors.Wrap(err, "failed to add auto-skip hash")
 	}
 	return nil
 }
+
+func (c *Client) AutoSkipPrune(ctx context.Context, org, project, pathPrefix, target string, deep bool) (int, error) {
+	target = strings.TrimPrefix(target, "+")
+
+	req := &askv.PruneTargetRequest{
+		OrgName:       org,
+		ProjectName:   project,
```

Comment:
I'm noticing that the project is specified here - isn't this supposed to be global within the organization?

Reply:
I'm not sure. I assumed we'd want to scope pruning to the project if passed via the flag or env. But that's not essential. If the value is empty in that request, it will be ignored.

Reply:
Shall we just explicitly pass it as empty string then? Just to make sure the hashes don't get sent to different places depending on whether `EARTHLY_PROJECT` is set or not.

I think that we've not yet figured out what to do with PROJECT, as that was a concept only needed for Earthly CI. I'd keep it global to the org until we get more clarity on that front.